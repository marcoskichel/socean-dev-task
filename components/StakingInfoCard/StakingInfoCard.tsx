import { Card, CardContent, styled, Typography, useTheme } from '@mui/material';
import InfoIcon from '@mui/icons-material/InfoOutlined';

const StyledCard = styled(Card)(({ theme }) => ({
  width: '100%',
  maxHeight: '19.25rem',
  backgroundColor: '#0D0D0D',
  [theme.breakpoints.up('md')]: {
    width: '19rem',
  },
}));

const StyledCardContent = styled(CardContent)(() => ({
  padding: '2rem',
  display: 'flex',
  flexDirection: 'column',
  gap: '1rem',
}));

const StakingInfoCard = () => {
  const theme = useTheme();

  return (
    <StyledCard>
      <StyledCardContent>
        <InfoIcon />
        <Typography variant="h6" color="primary">
          How does staking work?
        </Typography>
        <Typography variant="body1" color={theme.palette.text.secondary}>
          Staking allows users to participate in securing the network by locking
          up tokens. Consequently, users are rewarded for securing the network
          in the form of native tokens. The higher the amount of crypto-assets
          you pledge, the higher the rewards you receive. ... This means your
          crypto-assets earn while you sleep!
        </Typography>
      </StyledCardContent>
    </StyledCard>
  );
};

export default StakingInfoCard;
