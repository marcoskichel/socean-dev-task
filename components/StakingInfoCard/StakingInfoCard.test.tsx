import { render } from '@testing-library/react';
import StakingInfoCard from './StakingInfoCard';

describe('StakingInfoCard', () => {
  it('renders', () => {
    const { asFragment } = render(<StakingInfoCard />);
    expect(asFragment()).toMatchSnapshot();
  });
});
