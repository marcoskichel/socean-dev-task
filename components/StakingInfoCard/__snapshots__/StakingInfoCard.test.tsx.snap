// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StakingInfoCard renders 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-tg8ai1-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-39rnh6-MuiCardContent-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="InfoOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
        />
      </svg>
      <h6
        class="MuiTypography-root MuiTypography-h6 css-oblp1g-MuiTypography-root"
      >
        How does staking work?
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
      >
        Staking allows users to participate in securing the network by locking up tokens. Consequently, users are rewarded for securing the network in the form of native tokens. The higher the amount of crypto-assets you pledge, the higher the rewards you receive. ... This means your crypto-assets earn while you sleep!
      </p>
    </div>
  </div>
</DocumentFragment>
`;
